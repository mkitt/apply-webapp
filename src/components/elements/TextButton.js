// @flow
import glamorous from 'glamorous'
import { color, fontSize, space } from 'styled-system'
import { defaultProps } from 'recompose'
import { COLORS } from '../../constants/theme'

export default defaultProps({
  role: 'button',
})(glamorous.button(
  {
    position: 'relative',
    zIndex: 2,
    color: COLORS.BLUE,
    textDecoration: 'none',
    transition: 'color 200ms',
  },
  space,
  fontSize,
  color,
))
