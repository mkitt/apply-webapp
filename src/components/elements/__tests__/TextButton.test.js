import React from 'react'
import renderer from 'react-test-renderer'
import TextButton from '../TextButton'

describe('<TextButton />', () => {
  it('shows the correct snapshot tree for the <TextButton> element', () => {
    const component = <TextButton >The TextButton</TextButton>
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('shows the correct snapshot tree for the <TextButton> element with options', () => {
    const component = (
      <TextButton color="green" fontSize={18} mt={8}>
        The TextButton with options
      </TextButton>
    )
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
