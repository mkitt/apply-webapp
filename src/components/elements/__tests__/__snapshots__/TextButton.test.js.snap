// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextButton /> shows the correct snapshot tree for the <TextButton> element 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  z-index: 2;
  color: #35c5fc;
  text-decoration: none;
  transition: color 200ms;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<button
  className="css-0"
  role="button"
>
  The TextButton
</button>
`;

exports[`<TextButton /> shows the correct snapshot tree for the <TextButton> element with options 1`] = `
.css-0,
[data-css-0] {
  position: relative;
  z-index: 2;
  color: green;
  text-decoration: none;
  transition: color 200ms;
  margin-top: 512px;
  font-size: 18px;
  -webkit-transition: color 200ms;
  -moz-transition: color 200ms;
}

<button
  className="css-0"
  role="button"
>
  The TextButton with options
</button>
`;
